### ISSUE TITLE
Problèmes offline / recensement des bugs rencontrés

### ISSUE THREAD

**Issue by AudreyRemy** [GeotrekCE/Geotrek-rando-v3#867]
Bonjour,
Par ce ticket, nous aimerions tracer les échanges et les pistes à explorer pour remédier aux bugs rencontrés avec le mode offline. 
Pourriez-vous expliquer comment reproduire le bug en indiquant le contexte d'utilisation (quel navigateur, quel téléphone, quelle version de Geotrek-admin, de Geotrek-rando, quelles étapes réaliser pour reproduire le comportement obtenu, quelle url, etc.).

Pour les Pyrénées-Atlantiques : 
**Url** : https://nature64.fr/
**Navigateurs utilisé**s : Chrome (version 110.0.5481.65) et Firefox (version 110.0.1)  + « Application » installée via chrome.
**Téléphone** : Galaxy A51 – Version Android 13
**Geotren-rando V3** version 3.12.1
**Geotrek_admin**  version 2.94.0
**Fond de carte** : IGN (Pan V2)

Les données embarquées sont des itinérances (pédestre, VTT, cyclo, équestre).
En mode hors ligne, la page d’accueil est accessible ainsi que la page _Contenus hors lignes_.
Lorsqu’on sélectionne une fiche embarquée, une page d’erreur s’ouvre.

https://user-images.githubusercontent.com/78603108/220353346-49d5ca3b-5020-43ed-ae9f-f1d1cdddadab.mp4

**Comment by babastienne** [comment]
Je viens de tester, et à priori cela fonctionne en utilisant les mêmes conditions de navigation, sur nature64.fr.

Mon hypothèse est que les contenus embarqués hors connexion ont étés téléchargés avant la dernière mise à jour de Geotrek-Rando et que le cache n'est plus valide. Pour confirmer il faudrait tester la manipulation suivante :
- Dans la PWA, supprimer les éléments embarqués en hors connexion
- Re-télécharger en hors connexion une rando
- Vérifier qu'en mode avion cette rando est bien accessible

Si ça fonctionne alors il y a bien un problème sur le mécanisme d'invalidation du cache des contenus lors des mises à jours mais pas de problème applicatif en soit pour la version en elle-même.

**Comment by AudreyRemy** [comment]
@babastienne merci pour ce retour.
J'avais supprimé sur mon téléphone tout ce qui est _données_ et _cache_ des applications Chrome et Firefox. J'ai bien dû re-télécharger les rando en hors connexion.
2 collègues qui n'étaient jamais allés sur le site via leur téléphone (Samsung et Huawei) viennent de tester, ils ont le même problème que moi : la fiche est bien dans les _contenus hors ligne_ mais à l'ouverture ils obtiennent le message d'erreur.

**Comment by camillemonchicourt** [comment]
Moi j'avais testé et pas reproduis le soucis sur un Huawei P20.

**Comment by dtrucs** [comment]
Je repoduis le problème de @AudreyRemy, j'ai l'impression que ce problème apparait uniquement sur les trek "child" de grande itinérance.

En tout cas ça enregistre bien en cache et ça throw une erreur.

**Comment by dtrucs** [comment]
J'ai investigué, j'ai identifié en effet un problème avec les "enfants" de grande itinérance.

Si par exemple :
1/ Je navigue sur le lien suivant : https://nature64.fr/trek/12777-La-V-81-de-Biarritz-a-Urt?parentId=14941
2/ J'enregistre la rando.
3/ Je vais bien avoir le contenu de ma rando sauvegardé en cache avec la clé `https://nature64.fr/trek/12777-La-V-81-de-Biarritz-a-Urt?parentId=14941`.
4/ Je me met hors-ligne et je me rend dans ma liste de mes contenus enregistrés hors-ligne, je vais bien avoir dans ma liste la rando "La "V 81" de Biarritz à Urt".
5/ En observant le code, lors de la génération de la card, elle pointe vers le lien https://nature64.fr/trek/12777-La-V-81-de-Biarritz-a-Urt (sans le `?parentId=14941`)
6/ En cliquant dessus, l'application n'arrive pas à trouver une correspondance entre les 2 liens pour récupérer le contenu => crash de l'application.

Si je reprend mon ex et que j'ai enregistré la rando en n'étant pas dans la situation "enfant" de grande itinérance (= https://nature64.fr/trek/12777-La-V-81-de-Biarritz-a-Urt c'est ok de bout en bout).

En solution: lors de la sauvegarde d'une fiche, je propose de retirer les paramètres GET de l'URL courante (= tout ce qui est affiché après un ? dans l'url)

**Comment by AudreyRemy** [comment]
@dtrucs un immense merci pour cette investigation.
Pour la solution proposée, je ne suis pas compétente pour y répondre.

**Comment by camillemonchicourt** [comment]
> En solution: lors de la sauvegarde d'une fiche, je propose de retirer les paramètres GET de l'URL courante (= tout ce qui est affiché après un ? dans l'url)

Je sais pas si c'est la bonne idée, car si je sauvegarde offline une itinérance et ses étapes, alors on n'aura plus le lien entre l'itinérance et ses enfants pour lister les étapes de l'itinérance et passer d'une étape à l'autre ?

**Comment by dtrucs** [comment]
Effectivement, à vérifier mais je crois que les informations complémentaires liées à la grande itinérance ne sont pas sauvegardées en mode hors-ligne.

De plus cela mène une autre réflexion en terme fonctionnel : 
Il serait possible pour l'utilisateur d'enregistrer plusieurs fois une même rando : "seule" et en tant que "fille" d'une (ou plusieurs) grande itinérance. Il conviendrait alors d'améliorer la "card" pour signifier qu'elle est ou non une étape d'une grande itinerance, sans ça visuellement elles seront toutes identiques.

**Comment by JonathanALLAIN** [comment]
Bonjour,

Je reprends ce fil car nous avons depuis peu la V3 et donc la PWA (nous avons abandonné l'appli sur les stores).
En effet je reproduis l'erreur après téléchargement en offline d'une page d'erreur pour les fiches rando fille des itinérances. 
Que j'embarque en hors-ligne indépendamment la rando fille ou la mère, les filles ne s'ouvrent pas en offline... page d'erreur. 
Ce qui veut dire qu'aucune rando fille d'une itinérance ne peut s'ouvrir en hors-ligne.
Avez-vous des éléments de réponse ou correctif selon les réflexions de dtrucs ?
Merci ! Je me tiens à dispo pour tester si besoin.
